backend/de.metas.cucumber/src/test/resources/de/metas/cucumber/features/contract/contractTerm/cloneModularContractTerm.feature

```diff
@@ -0,0 +1,62 @@
+
+    And metasfresh contains ModCntr_Settings:
+      | ModCntr_Settings_ID.Identifier | Name                | M_Product_ID.Identifier | C_Calendar_ID.Identifier | C_Year_ID.Identifier |
+      | modCntr_settings_1             | Settings_17072023_1 | contract_module_product | harvesting_calendar      | y2022                |
+
+    And metasfresh contains ModCntr_Types:
+      | ModCntr_Type_ID.Identifier | Name                     | Value                    | Classname                                                               |
+      | modCntr_Types_1            | modCntr_Types_17072023_1 | modCntr_Types_17072023_1 | de.metas.contracts.modular.impl.PurchaseOrderLineModularContractHandler |
+
+    And metasfresh contains ModCntr_Modules:
+      | ModCntr_Module_ID.Identifier | SeqNo | Name                  | M_Product_ID.Identifier | InvoicingGroup | ModCntr_Settings_ID.Identifier | ModCntr_Type_ID.Identifier |
+      | modCntr_module_1             | 10    | moduleTest_17072023_1 | contract_module_product | Kosten         | modCntr_settings_1             | modCntr_Types_1            |
+
+    And metasfresh contains C_Flatrate_Conditions:
+      | C_Flatrate_Conditions_ID.Identifier | Name                     | Type_Conditions | OPT.OnFlatrateTermExtend | OPT.ModCntr_Settings_ID.Identifier | OPT.DocStatus |
+      | modularContractTerm_2022            | modularContractTerm_2022 | ModularContract | Ex                       | modCntr_settings_1                 | CO            |
+
+
+  @from:cucumber
+  Scenario: Happy flow - clone a Modular Contract Term -> new cloned Contract Term with Settings had the new harvest year
+  - Contract Term of type Modular Contract for harvest year 2022 already exists
+  - clone the contract for the harvest year 2023
+  - validate modular contract term cloned
+  - validate modular contract term can be edited (`Draft`)
+  - validate modular contract settings cloned for the harvest year 2023
+
+    When clone C_Flatrate_Conditions identified by modularContractTerm_2022 for year identified by y2023
+
+    Then validate cloned C_Flatrate_Conditions:
+      | C_Flatrate_Conditions_ID.Identifier | Name                          | Type_Conditions | OPT.OnFlatrateTermExtend | OPT.DocStatus |
+      | cloned_flatrate_conditions          | modularContractTerm_2022_2023 | ModularContract | Ex                       | DR            |
+
+    And validate cloned ModCntr_Settings had harvest year y2023
+
+
+  @from:cucumber
+  Scenario: Clone fail - for Modular Contract Terms with an existing harvest year
+  - Modular Contract term already exists  for harvest year 2022
+  - clone the Modular Contract for harvest year 2022
+  - fail with message "Einstellungen mit demselben Jahr sind bereits vorhanden"
+
+    Then fail with message Einstellungen mit demselben Jahr sind bereits vorhanden when clonning C_Flatrate_Conditions identified by modularContractTerm_2022 for year identified by y2022
```

Comment:
Same approach here. Please refactor this to use the same method here and at https://github.com/metasfresh/metasfresh/pull/15940/files#diff-a297012c86060d13df70ff268fcdc075f4e3e0a0958ae0b202621ab6a0bef4aeR47 ; use a data table for the common part, an identifier for the newly created settings and an optional error message.

This will allow to clean up the hardcoded "cloned_flatrate_conditions" garbage: https://github.com/metasfresh/metasfresh/pull/15940/files#diff-6cb6bcd718c2e18c4852da7e0af51a43d337e866a1693c58da69f4a98f97121bR240